### PR TITLE
[Common] Fix incorrect amax initialization in non-RHT NVFP4 C++ tests

### PR DESCRIPTION
**Pull request by Oleg-Goncharov**
# Description

This PR fixes the C++ test infrastructure for the non-RHT NVFP4 path.

Previously, the test flow populated the unused scale field instead of amax, which caused incorrect CPU/GPU comparisons in the non-RHT coverage. This change updates the test setup to initialize amax correctly.

Fixes # (issue)

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refactoring

## Changes

Please list the changes introduced in this PR:

- Fix incorrect amax initialization
- No kernel changes

# Checklist:

- [x] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [x] The functionality is complete
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/cpp/test_common.cu [hunk 1]
```diff
@@ -392,10 +395,14 @@ Tensor::Tensor(const std::string& name,
...
       if (scaling_mode == NVTE_NVFP4_1D_SCALING) {
         // Used for NVFP4 second stage scaling
-        cudaMalloc((void**)&scale, sizeof(float));  // NOLINT(*)
-        cudaMemset(scale, 0, sizeof(float));
-        scale_cpu_data_ = std::make_shared<float>(0);
-        tensor_.set_scale(scale, DType::kFloat32, std::vector<size_t>{1});
+        amax_cpu_data_ = std::make_shared<float>(0);
+        amax_cpu_data_columnwise_ = std::make_shared<float>(0);
+        cudaMalloc((void**)&amax, sizeof(float));  // NOLINT(*)
+        cudaMalloc((void**)&amax_columnwise, sizeof(float));  // NOLINT(*)
```
Comment (timmoon10) [inline tests/cpp/test_common.cu:401]: Both of these buffers will be leaked.

**Review by timmoon10 — DISMISSED** [review]
This is fine as a quick fix. I think the C++ testing infrastructure has gotten to the point where it is too messy to really trust. The recipe-specific logic is complicated and has many unhandled edge cases.

File: tests/cpp/test_common.cu (hunk 1 above)
Comment (Oleg-Goncharov) [reply tests/cpp/test_common.cu:401]: Oh, right!

**Review by timmoon10 — APPROVED** [review]
